### PR TITLE
Add focus

### LINKED ISSUE #30: Provide an easy way to run just one failing test
Perhaps this will be counter the philosophy of simplicity, but I've used several test suites which make "Run just this one test" easier than it currently is with minitest.cr.

Some examples in the wild:
- [minitest-focus](https://github.com/seattlerb/minitest-focus) provides a way with a focus statement which precedes test declarations
- rspec allows a syntax like this: `it "works", focus: true do`, and [allows tests to be otherwise tagged for grouping](http://rspec.info/documentation/3.7/rspec-core/RSpec/Core/ExampleGroup.html#it-class_method)
- `crystal spec` produces a runnable command as part of an error which will run just the specific failure, which is easy to copy and paste.

minitest.cr provides an `-n /name/` interface which is pretty close, but if tests are defined with `it "should pass"` the correct invocation is `-n /should_pass/` with an underscore added. Specifying `-n "should pass" will run no tests.

### PR DESCRIPTION
**Pull request by ysbaddaden**
Sometimes we want to focus on specific tests, but can't easily filter them with `-n`. The `focus` macro enables to mark a test method to be run, which will filter out other test methods for the test run. Specs can use the `focus: true` argument to achieve the same.

For example:

```crystal
require "minitest/autorun"
require "minitest/focus"

class MyTest < Minitest::Test
  focus def test_something
  end
end

describe "MyTest" do
  it "something", focus: true do
  end
end
```

closes #30

### CODE REVIEW

**Comment by robacarp** [comment]
Quick and simple, this works as advertised. Thank you @ysbaddaden

**Comment by robacarp** [comment]
Currently, when nothing is focused but `require "minitest/focus"` is present, no tests run. I'm not sure if this is intentional or an edge case that needs to be considered.

**Comment by robacarp** [comment]
Sorry for the spam. After looking through the pull, I believe the no-focus condition is unintentional. There are several ways around it, but I did this and it worked for me:

```diff
def should_run?(name)
-  focused_test?(name) && super(name)
+  if FOCUSES.any?
+    focused_test?(name) && super(name)
+  else
+    super(name)
+  end
end


```

File: src/focus.cr [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  class Test < Runnable
+    def should_run?(name)
+      focused_test?(name) && super(name)
+    end
+
+    def focused_test?(name)
+      {% begin %}
+      if focuses = FOCUSES[{{@type}}.name]?
+        focuses.empty? || focuses.includes?(name)
```
Comment (ysbaddaden) [inline src/focus.cr]: Damn, I missed an `else true` here.

**Comment by ysbaddaden** [comment]
Good catch. I just pushed a fix. Tell me if it works correctly now.

**Comment by robacarp** [comment]
@ysbaddaden it seems to work as expected. Thank you
